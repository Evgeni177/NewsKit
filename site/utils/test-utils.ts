import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
