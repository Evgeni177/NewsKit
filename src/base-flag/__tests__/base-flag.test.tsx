import {renderToFragmentWithTheme} from '../../test/test-utils';
import {BaseFlag} from '..';

describe('BaseFlag', () => {
  test('renders with default styles', () => {
    const fragment = renderToFragmentWithTheme(BaseFlag, {
      children: ['Text'],
    });
    expect(fragment).toMatchSnapshot();
  });
});
