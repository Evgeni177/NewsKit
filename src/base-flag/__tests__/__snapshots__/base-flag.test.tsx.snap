// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFlag renders with default styles 1`] = `
<DocumentFragment>
  .emotion-2 {
  box-sizing: border-box;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: default;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: calc(-4px/2);
  margin-right: calc(-4px/2);
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(4px/2);
  margin-right: calc(4px/2);
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-0"
      >
        Text
      </div>
    </div>
  </div>
</DocumentFragment>
`;
